nightguard/PredictionService.swift

```diff
@@ -0,0 +1,132 @@
+//
+//  PredictionService.swift
+//  nightguard
+//
+//  Created by Florian Preknya on 12/18/18.
+//  Copyright © 2018 private. All rights reserved.
+//
+
+import Foundation
+
+
+/// The brain that predicts some future BG values by applying a 2-degree
+/// polynomial regression on latest BG readings. Its prediction range is
+/// extended on 60 minutes from current time on.
+class PredictionService {
+    
+    static let shared = PredictionService()
```

Comment:
We should call it singleton since that's what it's called in the other Services. Or refractor its there if "shared" is more swifty ;-)

Reply:
I’ll change the name to Singleton to be consistent with the project naming convention.

Reply:
One more thing about project conventions I intended to ask you for a long time. Is there a reason why you didn’t use any third party libraries? 
I really miss some libraries that I’m using in my other projects, a lot of stuff could be more easily done and feature development time could be reduced significantly. Sometimes I decide not to start working on the project because I’ll have to fabric all the “bricks” by myself.  I use Cocoapods and Carthage (in my newer projects), both will do. What do you think?